inst/include/dust/distr/binomial.hpp

```diff
@@ -166,10 +166,12 @@ inline HOSTDEVICE real_t btrs(rng_state_t<real_t>& rng_state, int n_int, real_t
   return draw;
 }
 
+// NOTE: we return a real, not an int, as with deterministic mode this
+// will not necessarily be an integer
```

Comment:
Might need to be careful with that, as without changes in the odin model could lead to a non-integer n at the next step.
Rounding every step (especially if dt decreases) leads to issues. The thing we found was best with the rnorm based simulation was to keep the real at each dt step, and then round at the data steps.
I suppose this is an odin/model issue rather than anything that can be changed here (I think rounding the result would remove any flexibility around this issue)

Reply:
Looking at the wikipedia page, floor/ceil of np is the median

Reply:
Hmm, thanks for thinking about this. We *do* want non-integer n here, and I'll have to think about the way we handle that round, most likely moving it into the rbinom call

Reply:
Also in the limit of small step size, for this to work do you need to take a real for the input `n` too right (to get the mean)? Maybe another function/template to deal with the deterministic case would be best

Reply:
problem with a template is we then only have compile time polymorphism - I'll have a little think about a decent way of hiding the complexity, but fundamentally `rbinom` will need to change from `(int, real) -> int` to `(real, real) -> real` for this to work, and the round will need to move inside the nondeterministic arm